%YAML:1.0

#--------------------------------------------------------------------------------------------
# Camera Parameters. Adjust them!
#--------------------------------------------------------------------------------------------

# Camera calibration and distortion parameters (OpenCV) 
# 00-02
# Camera.fx: 718.856
# Camera.fy: 718.856
# Camera.cx: 607.1928
# Camera.cy: 185.2157

# 04-12
Camera.fx: 707.0912
Camera.fy: 707.0912
Camera.cx: 601.8873
Camera.cy: 183.1104

Camera.k1: 0.0
Camera.k2: 0.0
Camera.p1: 0.0
Camera.p2: 0.0


# 00-02
# Camera.width: 1241
# Camera.height: 376

# 04-12
Camera.width: 1226
Camera.height: 370


# Camera frames per second 
Camera.fps: 10.0

# stereo baseline times fx
Camera.bf: 386.1448

# Color order of the images (0: BGR, 1: RGB. It is ignored if images are grayscale)
Camera.RGB: 1

# Close/Far threshold. Baseline times.
ThDepth: 35

# baseline * fx
Stereo.bf: 386.1448 

#--------------------------------------------------------------------------------------------
# ORB Parameters
#--------------------------------------------------------------------------------------------

# ORB Extractor: Number of features per image
ORBextractor.nFeatures: 2000

# ORB Extractor: Scale factor between levels in the scale pyramid 	
ORBextractor.scaleFactor: 1.2

# ORB Extractor: Number of levels in the scale pyramid	
ORBextractor.nLevels: 8

# ORB Extractor: Fast threshold
# Image is divided in a grid. At each cell FAST are extracted imposing a minimum response.
# Firstly we impose iniThFAST. If no corners are detected we impose a lower value minThFAST
# You can lower these values if your images have low contrast			
ORBextractor.iniThFAST: 20
ORBextractor.minThFAST: 7


# data_sequence: "/home/kinggreat24/Documents/test_data/kitti/00"
data_sequence: "/media/kinggreat24/Samsung_T5/data/kitti_data_full/odometry/unzip/data/dataset/sequences/07"

# lidar 2 camera 00-02
# extrinsicMatrix: !!opencv-matrix
#    rows: 4
#    cols: 4
#    dt: d
#    data: [ 0.0004276802385584, -0.9999672484946, -0.008084491683471, -0.01198459927713,
#           -0.007210626507497,  0.008081198471645, -0.9999413164504, -0.05403984729748,
#            0.9999738645903,  0.0004859485810390, -0.007206933692422, -0.2921968648686,
#            0,0,0,1]

# lidar 2 camera 04-12
extrinsicMatrix: !!opencv-matrix
   rows: 4
   cols: 4
   dt: d
   data: [-0.001857739385241, -0.9999659513510, -0.008039975204516, -0.004784029760483,
          -0.006481465826011,  0.008051860151134, -0.9999466081774, -0.07337429464231,
           0.9999773098287, -0.001805528627661, -0.006496203536139, -0.3339968064433,
           0,0,0,1]


# Tracker Setting
# ScaleEstimator : None, TDistributionScale
# WeightFunction attribute : None, TDistributionWeight
# max leve : seq2 = 1, seq3 = 3->0, seq12 = 1, seq20 = 0, seq21 = 1
Tracker.levels: 3
Tracker.min_level: 0
Tracker.max_level: 2
Tracker.max_iteration: 100
Tracker.scale_estimator: "TDistributionScale"
Tracker.weight_function: "TDistributionWeight"

# 特征法相关参数
FTracker.scale_estimator: "MADScale"
FTracker.weight_function: "HuberWeight"
# FTracker.scale_estimator: "None"
# FTracker.weight_function: "None"